### PR TITLE
Bug Fixes - ENG-7596 | ENG-7469 | ENG-7468 | ENG-7466

### PR DESCRIPTION
**Pull request by oakinh**
gp-api: https://github.com/thegoodparty/gp-admin/pull/68
gp-webapp: https://github.com/thegoodparty/gp-webapp/pull/1777

https://goodparty.clickup.com/t/90132012119/ENG-7596
https://goodparty.clickup.com/t/90132012119/ENG-7469
https://goodparty.clickup.com/t/90132012119/ENG-7468
https://goodparty.clickup.com/t/90132012119/ENG-7466

---

> [!NOTE]
> **Medium Risk**
> Touches authentication-adjacent user identity enrichment and can now persist `clerkId` via lazy linking, which could affect how users are resolved and displayed if Clerk data is missing or mismatched.
> 
> **Overview**
> Improves Clerk→DB user enrichment to **avoid overwriting local identity fields with empty Clerk values** and to **never serve stale DB avatars** when Clerk is unreachable, missing, or the user has no `clerkId`.
> 
> `ClerkUserEnricherService` now supports **lazy linking**: when a user has a DB `id` + `email` but no `clerkId`, it attempts an email lookup in Clerk, persists the resolved `clerkId` (with negative caching and unique-constraint handling), then enriches fields. `SessionGuard.resolveUser` is updated to mirror these rules (fall back to DB values for empty Clerk strings and null out avatar when Clerk can’t be reached).
> 
> Adds two **one-time Prisma/Clerk backfill scripts** with `--dry-run` + JSONL audit logs: `backfillUserClerkIds.ts` (populate missing `clerkId` by email) and `backfillUserNamesFromClerk.ts` (sync `firstName`/`lastName`/`name` for users with `clerkId`). Includes a new test suite covering these enrichment edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0b3c038e67e7e728ce04f54ebcce393e9fe669c3. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by tomer-tgp** [comment]
Task linked: [ENG-7466 User's Display Name Truncates or Shows Incorrect Value During Impersonation](https://app.clickup.com/t/90132012119/ENG-7466)
Task linked: [ENG-7468 User Avatar in Admin Tool Loads from Local DB Instead of Clerk](https://app.clickup.com/t/90132012119/ENG-7468)
Task linked: [ENG-7469 [GP-ADMIN] - User and Campaign ID Fields need specific Labels in Admin Tool](https://app.clickup.com/t/90132012119/ENG-7469)
Task linked: [ENG-7591 Switching Offices Should Wipe DistrictOverrideId](https://app.clickup.com/t/90132012119/ENG-7591)

File: src/vendors/clerk/services/clerk-user-enricher.service.ts [hunk 1]
```diff
@@ -160,13 +165,115 @@ export class ClerkUserEnricherService {
...
+      if (!clerkUser?.id) {
+        this.emailMissCache.set(key, Date.now() + this.cacheTtlMs)
+        return null
+      }
+
+      await this.prisma.user.update({
+        where: { id },
+        data: { clerkId: clerkUser.id },
+      })
+      return clerkUser.id
```
[the remark this answers is not in the document]
Comment (oakinh) [reply src/vendors/clerk/services/clerk-user-enricher.service.ts:263]: https://github.com/thegoodparty/gp-api/pull/1539/commits/2d7a16e07f09d6133b2c86ce94ca4dc2f127962d

**Comment by tomer-tgp** [comment]
Task linked: [ENG-7596 Impersonated Users w/o an org should be redirected to onboarding](https://app.clickup.com/t/90132012119/ENG-7596)

**Review by tomer-tgp — APPROVED** [review]
